runtime/compiler/env/CHTable.cpp

```diff
@@ -991,6 +998,115 @@ bool CollectCompiledImplementors::visitSubclass(TR_PersistentClassInfo *cl)
+
+   virtual bool visitSubclass(TR_PersistentClassInfo *cl) override;
+   void cacheResolvedMethods();
+
+private:
+   std::vector<TR_OpaqueClassBlock *> _subClasses;
+   };
+
+bool
+CollectResolvedImplementors::visitSubclass(TR_PersistentClassInfo *cl)
+   {
+   // must only be used on JITServer server-side
+   if (!comp()->isOutOfProcessCompilation())
+      {
+      stopTheWalk();
+      return false;
+      }
+
+   auto callerMethod = static_cast<TR_ResolvedJ9JITServerMethod *>(_callerMethod);
+   TR_ResolvedMethodType type = isInterface() ?
+      TR_ResolvedMethodType::Interface :
+      TR_ResolvedMethodType::VirtualFromOffset;
+   auto compInfoPT = static_cast<TR::CompilationInfoPerThreadRemote *>(static_cast<TR_J9VMBase *>(fe())->_compInfoPT);
+
+   // verify that our subclass meets all conditions
+   TR_OpaqueClassBlock *classId = cl->getClassId();
+   if (!TR::Compiler->cls.isAbstractClass(comp(), classId)
+      && !TR::Compiler->cls.isInterfaceClass(comp(), classId))
+      {
+      ++_numVisitedSubClasses;
+      if (_numVisitedSubClasses > _maxNumVisitedSubClasses)
+         {
+         // set count greater than maxCount, to indicate failure and force exit
+         _count = _maxCount + 1;
+         stopTheWalk();
+         return false;
+         }
+
+      // if the implementor hasn't been cached yet,
+      // add the corresponding subclass to the list
+      TR_ResolvedMethod *resolvedMethod;
+      TR_ResolvedMethodKey key =
+         compInfoPT->getResolvedMethodKey(
+            type,
+            reinterpret_cast<TR_OpaqueClassBlock *>(callerMethod->constantPoolHdr()),
+            _slotOrIndex,
+            classId);
+      if (!compInfoPT->getCachedResolvedMethod(
+             key,
+             callerMethod,
+             &resolvedMethod))
+         {
+         _subClasses.push_back(classId);
+         }
+      else if (resolvedMethod)
+         {
+         // check to see if there are any duplicates
+         int32_t i;
+         for (i = 0; i < _count; i++)
+            if (resolvedMethod->isSameMethod(_implArray[i]))
+               break;  // we already listed this method
+         if (i >= _count) // brand new implementer
+            {
+            _implArray[_count++] = resolvedMethod;
+            if (_count >= _maxCount)
+               {
+               stopTheWalk();
+               }
+            }
+         }
+      }
+
+   return true;
+   }
+
+void
+CollectResolvedImplementors::cacheResolvedMethods()
+   {
+   // must only be used on JITServer server-side
+   // _count >= _maxCount indicates that either all required implementors
+   // are already cached or that too many subclasses were visited.
+   // Remote call not needed.
+   if (!comp()->isOutOfProcessCompilation() || _count >= _maxCount)
+      {
+      return;
+      }
+
+   static_cast<TR_ResolvedJ9JITServerMethod *>(_callerMethod)->cacheImplementorMethods(
+      _subClasses,
+      _slotOrIndex,
+      isInterface(),
+      2);
+   }
+#endif
```

Comment:
Add /* defined(J9VM_OPT_JITSERVER)  */

Reply:
I'll also add it in other places in this file that don't have it.